contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -323,6 +323,10 @@ tsql_cast_raw_info_t tsql_cast_raw_infos[] =
 	{TSQL_CAST_WITHOUT_FUNC_ENTRY, "pg_catalog", "bytea", "pg_catalog", "text", NULL, 'i', 'i'},
 	{TSQL_CAST_WITHOUT_FUNC_ENTRY, "sys", "datetimeoffset", "pg_catalog", "text", NULL, 'i', 'i'},
 	{TSQL_CAST_WITHOUT_FUNC_ENTRY, "pg_catalog", "time", "pg_catalog", "text", NULL, 'i', 'i'},
+/*  date/time -> string via I/O */
+	{TSQL_CAST_WITHOUT_FUNC_ENTRY, "pg_catalog", "time", "sys", "varchar", NULL, 'i', 'i'},
+	{TSQL_CAST_WITHOUT_FUNC_ENTRY, "pg_catalog", "date", "sys", "varchar", NULL, 'i', 'i'},
+	{TSQL_CAST_WITHOUT_FUNC_ENTRY, "sys", "datetimeoffset", "sys", "varchar", NULL, 'i', 'i'},
```

Comment:
Ok, so we have added these casts so the functions will work the source datatype. Please check for the other types as well.

Reply:
Only these 3 IMPLICIT Casts to VARCHAR were missing rest all are already there according to TSQL. Also tests are there to verify the same.

Reply:
Earlier, we didn't allow implicit casts from `date, time and datetimeoffset` to `varchar` whereas TSQL does, Although we already have casts from these functions to `TEXT` without function, so I don't think there would any issue here, but in future if there is need for adding functions for these casts then we can do it as a part of gradual development.